README.md

```diff
@@ -1,8 +1,53 @@
-Telemetry Services
-------------------
+# Mozilla Pipeline Schemas
 
-This repository contains schemas for Mozilla Firefox Telemetry pings.
+This repository contains schemas for Mozilla's data ingestion pipeline and data
+lake outputs.
 
-These schemas are used to validate incoming submissions at ingestion time. The [RapidJSON](http://rapidjson.org/md_doc_schema.html) library is used for JSON Schema Validation. This has implications for what kinds of string patterns are supported, see the `Conformance` section in the linked document for further details.
+The JSON schemas are used to validate incoming submissions at ingestion time.
+The [RapidJSON](http://rapidjson.org/md_doc_schema.html) library is used for
+JSON Schema Validation. This has implications for what kinds of string patterns
+are supported, see the `Conformance` section in the linked document for further
+details.
 
-To learn more about writing JSON Schemas, [Understanding JSON Schema](https://spacetelescope.github.io/understanding-json-schema/index.html) is a great resource.
+To learn more about writing JSON Schemas,
+[Understanding JSON Schema](https://spacetelescope.github.io/understanding-json-schema/index.html)
+is a great resource.
+
+The [Parquet-MR](https://github.com/apache/parquet-format/blob/master/LogicalTypes.md)
+schemas are used for direct to parquet output.
+
+To learn more about Parquet-MR Schemas.
+[Parquet Schema Examples](https://mozilla-services.github.io/lua_sandbox_extensions/parquet/io_modules/lpeg/parquet.html)
+is a limited resource as the full specification is only documented in code.
+
+## Build
+
+### Note
+
+All schemas are generated from the 'templates' directory and written into the
+'schemas' directory (i.e., the artifacts are generated/saved back into the
+repository as per mreid's request).
```

Comment:
The general requirements for the templated approach, as I see them:

- Ability to factor out common schema components (telemetry "environment" section, clientId / docId spec, etc).
- One source schema can be used to generate multiple output schemas. Example: `main` schema should be used for both `main` and `saved-session` doctypes.
- Generate other schema formats from canonical JSONSchema representation (namely parquet-mr schema representation, but possibly others in the future).
- Produce standalone single-document JSON Schemas. No external `$ref`s. This is to avoid the need to implement custom ref resolvers in multiple languages/platforms.
- Simplify the workflow for adding a new schema, particularly if it shares common components.

Regarding storing generated artifacts in the repository, the rationale is twofold. 

First, it lets us easily see and refer to complete schemas as they are actually used. This means that the schemas can be referenced directly in bugs and such, as well as being fetched directly from the repo for testing other schema consumers (*test* being important here, as any production use should be using the installable packages). 

Second, it gives us a changelog for each schema, rather than having to reason about changes to templated external pieces and when/how that impacted a given doctype's schema over time. This means that it should be easy to look back in time for the provenance of different parts of the schema for each doc type.